fixham/pkg/api/builder.go

```diff
@@ -0,0 +1,72 @@
+package api
+
+import (
+	"os"
+
+	"github.com/projectcalico/calico/fixham/internal/config"
+	"github.com/projectcalico/calico/fixham/internal/docker"
+	"github.com/projectcalico/calico/fixham/pkg/goyek"
+)
+
+// Builder is a struct that represents a component of the project.
+type Builder struct {
+	config *config.Config
+	tasks  map[string]*goyek.GoyekTask
+}
+
+// NewBuilder returns a new Builder
+func NewBuilder() *Builder {
+	return &Builder{
+		config: config.LoadConfig(),
+		tasks:  make(map[string]*goyek.GoyekTask),
+	}
+}
+
+// Path returns the path used for the component
+func (c *Builder) Path() string {
+	currentDir, _ := os.Getwd()
+	return currentDir
+}
+
+// Name returns the name of the component
+func (c *Builder) Name() string {
+	return c.config.Name
+}
+
+// PackageName returns the package name of the component
+func (c *Builder) PackageName() string {
+	return c.config.PackageName
+}
+
+// Config returns the configuration of the component
+func (c *Builder) Config() *config.Config {
+	return c.config
+}
+
+// DockerRunner returns a DockerRunner to be used in the component
+func (c *Builder) DockerRunner() *docker.DockerRunner {
+	runner := docker.MustDockerRunner()
+	return runner
+}
+
+// DockerGoBuildRunner returns a GoBuildRunner to be used in the component
+func (c *Builder) DockerGoBuildRunner() *docker.GoBuildRunner {
```

Comment:
Same for a number of these functions - I think this "Builder" concept may be less useful now that we are scoping this to be a release / hashrelease tool instead of a generic framework for building and running tasks. Do we need it at all?

Reply:
this function will still be needed once we start to optimize the current logic

Reply:
@radTuti could you give an example of the optimizations you have in mind that will need this?

I very well could be misremembering, but I recall discussing that this wasn't needed last time we hopped on Zoom.

I would rather we leave it out of this PR if it isn't being used. We can always add it in another PR if there is a use for it, but dead code / functions that aren't called shouldn't be merged.

Reply:
I will take it out since there is no immediate usage.

I was hoping the gen-version work for operator but I will need to re-think that implentation ATM